mllib/src/main/scala/org/apache/spark/ml/classification/DecisionTreeClassifier.scala

```diff
@@ -94,17 +94,33 @@ final class DecisionTreeClassifier @Since("1.4.0") (
     }
     val oldDataset: RDD[LabeledPoint] = extractLabeledPoints(dataset)
     val strategy = getOldStrategy(categoricalFeatures, numClasses)
+
+    val instr = Instrumentation.create(this, oldDataset)
+    instr.logParams(params: _*)
+
     val trees = RandomForest.run(oldDataset, strategy, numTrees = 1, featureSubsetStrategy = "all",
       seed = $(seed), parentUID = Some(uid))
-    trees.head.asInstanceOf[DecisionTreeClassificationModel]
+
+    instr.logNumFeatures(oldDataset.first().features.size)
```

Comment:
Doing it here causes some potential performance issues, because you materialize the 1st data partition but you do not cache it. I think it will be easier to log this in `RandomForest.run()` (and pass the instrumentation as an extra argument), so that this is done after `baggedInput`